src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1587,8 +1579,20 @@ tx_gateway_frontend::do_commit_tm_tx(
         ok = ok && (r.ec == tx_errc::none);
     }
     if (!ok) {
+        if (is_transaction_ga()) {
+            outcome->set_value(tx_errc::unknown_server_error);
+        }
         co_return tx_errc::unknown_server_error;
     }
+
+    if (is_transaction_ga()) {
```

Comment:
this isn't needed, it's ok to encounter a failing commit after we acked to the client because it fails only in case there is a conflicting client which rolled past transaction forward so we don't lie to clients even if commit fails